bertopic/plotting/_barchart.py

```diff
@@ -46,17 +48,24 @@ def visualize_barchart(topic_model,
     colors = itertools.cycle(["#D55E00", "#0072B2", "#CC79A7", "#E69F00", "#56B4E9", "#009E73", "#F0E442"])
 
     # Select topics based on top_n and topics args
-    freq_df = topic_model.get_topic_freq()
-    freq_df = freq_df.loc[freq_df.Topic != -1, :]
+    df_topic_info = topic_model.get_topic_info()
+    df_topic_info = df_topic_info.loc[df_topic_info.Topic != -1, :]
+    topic_label_map = {
+        df_topic_info.Topic.iloc[i] : df_topic_info.CustomName.iloc[i]
+        for i in range(len(df_topic_info))
+    }
     if topics is not None:
         topics = list(topics)
     elif top_n_topics is not None:
-        topics = sorted(freq_df.Topic.to_list()[:top_n_topics])
+        topics = sorted(df_topic_info.Topic.to_list()[:top_n_topics])
     else:
-        topics = sorted(freq_df.Topic.to_list()[0:6])
+        topics = sorted(df_topic_info.Topic.to_list()[0:6])
```

Comment:
I believe this is not necessary. I would advise reverting this code to what it was before and creating the `subplot_titles` a bit differently as I will show below. That way, the structure of selecting topics is similar to other visualization options and it minimizes the code needed to use the `custom_labels`.